client/gutenberg/editor/edit-post/components/header/header-toolbar/index.js

```diff
@@ -78,7 +78,7 @@ function getCloseButtonPath( routeHistory, site ) {
 	// @see post-editor/editor-ground-control/index.jsx
 	const lastNonEditorPath = findLast(
 		routeHistory,
-		action => ! action.path.match( editorPathRegex )
+		( { path } ) => '/gutenberg' !== path && ! path.match( editorPathRegex )
```

Comment:
looks like we might make some clearer sense if we just fix the RegExp pattern. I was going to do this but it wasn't clear to me what the goals were to match…

**What's there already**
```js
^\/(?:gutenberg\/)?(?:page|post|edit\/[^/\n]+)(?:\/|$)
```
<img width="234" alt="screen shot 2018-12-04 at 2 49 20 pm" src="https://user-images.githubusercontent.com/5431237/49475197-cdefb900-f7d3-11e8-82ca-c551cd6680d1.png">

**Also allowing just the `/gutenberg` path, fixing some possible mismatches**
```js
^(?:(?:\/gutenberg)?\/(?:page|post|edit\/[^/\n]+)(?:\/|$)|(?:\/gutenberg\/?)$)
```
<img width="231" alt="screen shot 2018-12-04 at 3 04 25 pm" src="https://user-images.githubusercontent.com/5431237/49475970-ecef4a80-f7d5-11e8-848c-096b01359ea4.png">

Mostly I'm confused on which paths we want to match and which we don't. The RegExp as written leaves some cases open to speculation. Could we clarify which of these examples we want and which we don't want if my pictures are wrong?

---

The easiest way I thought of to put together the match which says it could be (a) `/gutenberg` or (b) `/gutenberg/(post|page|edit…)` or (c) `/(post|page|edit…)` was to add just a wee bit of duplication into the RegExp (we could have eliminated it with a back reference) and try the full `gutenberg?/(post|page|edit…)` section first, failing if that doesn't match to the plain `/gutenberg` routes. if we try and do it with `/gutenberg?` first then it will either (d) match paths that are like `/gutenberg/pasta` or (e) fail to match `/gutenberg/post` because it first matches the shorter segment